From 3d845c38b0620547a58654a3b38ceed483b9779f Mon Sep 17 00:00:00 2001
From: N Balachandran <nbalacha@redhat.com>
Date: Wed, 14 Mar 2018 10:05:06 +0530
Subject: [PATCH 221/236] cluster/dht:  Skipped files are not treated as errors

For skipped files, use a return value of 1 to prevent
error messages being logged.
upstream patch: https://review.gluster.org/#/c/19710/

> Change-Id: I18de31ac1a64d4460e88dea7826c3ba03c895861
> BUG: 1553598
> Signed-off-by: N Balachandran <nbalacha@redhat.com>

Change-Id: I18de31ac1a64d4460e88dea7826c3ba03c895861
BUG: 1546941
Signed-off-by: N Balachandran <nbalacha@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/132609
Tested-by: RHGS Build Bot <nigelb@redhat.com>
Reviewed-by: Atin Mukherjee <amukherj@redhat.com>
---
 xlators/cluster/dht/src/dht-rebalance.c | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/xlators/cluster/dht/src/dht-rebalance.c b/xlators/cluster/dht/src/dht-rebalance.c
index a4be348..eee00b8 100644
--- a/xlators/cluster/dht/src/dht-rebalance.c
+++ b/xlators/cluster/dht/src/dht-rebalance.c
@@ -594,7 +594,7 @@ __check_file_has_hardlink (xlator_t *this, loc_t *loc,
                                 "Migration skipped for:"
                                 "%s: file has hardlinks", loc->path);
                         *fop_errno = ENOTSUP;
-                        ret = -1;
+                        ret = 1;
                 }
        }
 
@@ -649,7 +649,7 @@ __is_file_migratable (xlator_t *this, loc_t *loc,
                                 "Migrate file failed: %s: File has locks."
                                 " Skipping file migration", loc->path);
                         *fop_errno = ENOTSUP;
-                        ret = -1;
+                        ret = 1;
                         goto out;
                 }
         }
@@ -964,7 +964,7 @@ __dht_check_free_space (xlator_t *this, xlator_t *to, xlator_t *from, loc_t *loc
                         /* this is not a 'failure', but we don't want to
                            consider this as 'success' too :-/ */
                         *fop_errno = ENOSPC;
-                        ret = -1;
+                        ret = 1;
                         goto out;
                 }
         }
@@ -2719,7 +2719,7 @@ gf_defrag_migrate_single_file (void *opaque)
 
         ret = dht_migrate_file (this, &entry_loc, cached_subvol,
                                 hashed_subvol, rebal_type, &fop_errno);
-        if (ret < 0) {
+        if (ret == 1) {
                 if (fop_errno == ENOSPC) {
                         gf_msg_debug (this->name, 0, "migrate-data skipped for"
                                       " %s due to space constraints",
@@ -2768,8 +2768,12 @@ gf_defrag_migrate_single_file (void *opaque)
                                 DHT_MSG_MIGRATE_FILE_SKIPPED,
                                 "File migration skipped for %s.",
                                 entry_loc.path);
+                }
+
+                ret = 0;
 
-                } else if (fop_errno != EEXIST) {
+        } else if (ret < 0) {
+                if (fop_errno != EEXIST) {
                         gf_msg (this->name, GF_LOG_ERROR, fop_errno,
                                 DHT_MSG_MIGRATE_FILE_FAILED,
                                 "migrate-data failed for %s", entry_loc.path);
-- 
1.8.3.1

